src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl_SourceGenerators.cs

```diff
@@ -70,9 +69,8 @@ public void SubscribeToSourceGeneratorImpactingEvents()
     }
 
     [Export(typeof(ICommandHandler))]
-    [ContentType(ContentTypeNames.RoslynContentType)]
-    [ContentType(ContentTypeNames.XamlContentType)]
     [Name(PredefinedCommandHandlerNames.SourceGeneratorSave)]
+    [ContentType(StandardContentTypeNames.Text)]
```

Comment:
>[ContentType(StandardContentTypeNames.Text)] [](http://example.com/codeflow?start=4&length=44)

I don't have any suggestions here, but won't this cause roslyn to load when opening txt files?

Reply:
bleagh.

@jasonmalinowski @sharwell  any ideas on this?

Reply:
@olegtk  for thoughts as well.  We only want to listen to saves when roslyn is actually loaded.  but we do need to know about saves to files that aren't roslyn content type (like saving an xml file inside a roslyn project that will then have an impact on generators).

Reply:
Is RDT or another vs api a possibility here?

Reply:
Yeah listening to the RDT or some other VS would probably be what we need to do here.

Reply:
@jasonmalinowski anything i can look into?

Reply:
@CyrusNajmabadi maybe https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/ProjectSystem/OpenTextBufferProvider.cs#L32

Looks like there is an OnAfterSave method we could implement